node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -427,9 +430,15 @@ void tick() {
 
     // Public for testing
     void converge() {
-        final ContainerNodeSpec nodeSpec = nodeRepository.getContainerNodeSpec(hostname)
-                .orElseThrow(() ->
-                        new IllegalStateException(String.format("Node '%s' missing from node repository.", hostname)));
+        final Optional<ContainerNodeSpec> nodeSpecOptional = nodeRepository.getContainerNodeSpec(hostname);
+
+        // We just removed the node from node repo, so this is expected until NodeAdmin stop this NodeAgent
+        if (!nodeSpecOptional.isPresent() && expectNodeNotInNodeRepo) return;
```

Comment:
There should probably be a warning for this?

Reply:
_Maybe_ INFO, why would there be a warning for something that is expected and not a problem?

Reply:
ok, I guess it's fine.